app/views.py

```diff
@@ -0,0 +1,390 @@
+    """ One bucket list
+            -view one
+            -update one
+            -delete one"""
+
+    def get(self, bucketlist_id):
+        """ View single bucketlist"""
+        user_id = g.user.user_id
+        bucketlist = BucketList.query.filter_by(
+            bucketlist_id=bucketlist_id, created_by=user_id).first()
+
+        # check if theres a bucketlist
+        if bucketlist:
+            return marshal(bucketlist, format_bucketlist), 200
+            # status code - ok
+
+        return {'error': "BucketList with id {} not found.".format(bucketlist_id)}, 404
+        # status code - not found
+
+    def put(self, bucketlist_id):
+        """ Update a bucketlist"""
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('title', type=str, location='json')
+        self.reqparse.add_argument('description', location='json')
+
+        user_id = g.user.user_id
+        args = self.reqparse.parse_args()
+
+        # testing if the bucketlist with that id exists for this user
+        bucketlist = BucketList.query.filter_by(bucketlist_id=bucketlist_id,
+                                                created_by=user_id).first()
+
+        if not bucketlist:
+            return {'error': 'Bucketlist with id {} not found'.format(bucketlist_id)}, 404
+
+        if args.title:
+            bucketlist.title = args.title
+        if args.description:
+            bucketlist.description = args.description
+        db.session.commit()
+        return {'message': 'Bucketlist with id {} was updated'.format(bucketlist_id)}
+
+    def delete(self, bucketlist_id):
+        """ Delete a bucketlist"""
+        user_id = g.user.user_id
+        bucketlist = BucketList.query.filter_by(
+            bucketlist_id=bucketlist_id, created_by=user_id).first()
+
+        # check if theres a bucketlist
+        if bucketlist:
+            db.session.delete(bucketlist)
+            db.session.commit()
+            # 204
+            return {'message': 'The bucketlist with ID %s was deleted' % bucketlist_id}
+            # status code - request processed, no content returned
+
+        return {'error': "Bucketlist with id {} not found.".format(bucketlist_id)}, 404
+        # status code - not found
+
+
+class BucketListsAPI(Resource):
+    """create one or view many
+    root of bucketist"""
+    decorators = [auth.login_required]
+
+    def post(self):
+        """ New Bucketlist"""
+        self.reqparse = reqparse.RequestParser()
+        self.reqparse.add_argument('title', type=str, required=True,
+                                   help="Task title can't be blank", location='json')
+        self.reqparse.add_argument(
+            'description', type=str, default="", location='json')
+
+        args = self.reqparse.parse_args()
+        title = args['title']
+        description = args['description']
+        user_id = g.user.user_id
+
+        # check if title is null
+        if title == "":
```

Comment:
@neddoz I realized that reqparse only checks if the parameter has been passed in the request, but doesn't check if its actually an empty string.